CRM/Remoteevent/RegistrationProfile.php

```diff
@@ -269,8 +429,137 @@ function(int $carry, string $item) {
                 }
             }
         }
+
+        // Validate price fields.
+        if ((bool) $event['is_monetary']) {
+            foreach ($this->validatePriceFields(
+                $event,
+                $data,
+                $additionalParticipantsCount,
+                $l10n
+            ) as $field_name => $error) {
+                $validationEvent->addValidationError($field_name, $error);
+            }
+        }
     }
 
+  /**
+   * @phpstan-param array<string, mixed> $event
+   * @phpstan-param array<string, mixed> $submission
+   *
+   * @phpstan-return array<string, string>
+   *   An array with field names as keys and corresponding localised error
+   *   messages as values.
+   * @throws \CRM_Core_Exception
+   */
+  protected function validatePriceFields(
```

Comment:
What about moving the code for price field validation to an own class?

Reply:
As the definition of price fields in `\CRM_Remoteevent_RegistrationProfile::getProfilePriceFields()` might be subject to customizations in sub classes, validation needs to be adjustable as well, so I'd keep it in the profile class for easy access via `parent::validatePriceFields()` in overridden methods.

Reply:
The implementation itself might still live at another place and could be called from here...